### PR TITLE
Error traceability in CARL

### PR DESCRIPTION
**Pull request by Lenita-bu**

### CODE REVIEW

File: opendut-util/src/logging.rs [hunk 1]
```diff
@@ -46,16 +55,85 @@ pub fn initialize_with_config(config: LoggingConfig) -> Result<(), Error> {
...
+    let opentelemetry_layer = config.opentelemetry_endpoint.clone().map(|endpoint| {
+        let otlp_exporter = opentelemetry_otlp::new_exporter()
+            .tonic().with_endpoint(endpoint.url);
+        let tracer = opentelemetry_otlp::new_pipeline()
+            .tracing()
+            .with_exporter(otlp_exporter)
+            .with_trace_config(
+                opentelemetry_sdk::trace::config().with_resource(Resource::new(vec![KeyValue::new(
+                    opentelemetry_semantic_conventions::resource::SERVICE_NAME,
+                    "OTel",
```
Comment (kKdH) [inline opendut-util/src/logging.rs:68]: Introduce config parameter for the `SERVICE_NAME` value.

File: opendut-vpn/opendut-vpn-netbird/src/lib.rs
Comment (kKdH) [inline opendut-vpn/opendut-vpn-netbird/src/lib.rs:1]: Add `tracing` macro to all functions __without__ `name` parameter.

- `create_cluster`
- `delete_cluster`
- `create_peer`
- `generate_vpn_peer_configuration`

File: opendut-vpn/opendut-vpn-netbird/src/lib.rs [hunk 2]
```diff
@@ -69,6 +69,7 @@ impl NetbirdManagementClient {
 #[async_trait]
 impl VpnManagementClient for NetbirdManagementClient {
 
+    #[tracing::instrument(name = "create_cluster", skip(self), level="trace")]
```
Comment (kKdH) [inline opendut-vpn/opendut-vpn-netbird/src/lib.rs:72]: remove `name` parameter

File: opendut-vpn/opendut-vpn-netbird/src/lib.rs [hunk 3]
```diff
@@ -102,6 +103,7 @@ impl VpnManagementClient for NetbirdManagementClient {
         Ok(())
     }
 
+    #[tracing::instrument(name = "delete_cluster", skip(self), level="trace")]
```
Comment (kKdH) [inline opendut-vpn/opendut-vpn-netbird/src/lib.rs:106]: remove `name` parameter

**Review by kKdH — CHANGES_REQUESTED** [review]
Thank you for your contribution!

File: opendut-carl/carl.toml [hunk 4]
```diff
@@ -33,3 +33,7 @@ auth.type = ""
 auth.secret = ""
 timeout.ms = 10000
 retries = 5
+
+[monitoring.opentelemetry]
```
[the remark this answers is not in the document]
Comment (Lenita-bu) [reply opendut-carl/carl.toml:37]: Removed monitoring

File: opendut-carl/carl.toml [hunk 5]
```diff
@@ -33,3 +33,7 @@ auth.type = ""
...
+enabled = true
+endpoint = "http://localhost:4317"
```
[the remark this answers is not in the document]
Comment (Lenita-bu) [reply opendut-carl/carl.toml:39]: The values in carl-development.toml are set as required.

File: opendut-util/src/logging.rs [hunk 6]
```diff
@@ -46,16 +55,85 @@ pub fn initialize_with_config(config: LoggingConfig) -> Result<(), Error> {
...
 #[derive(Default)]
 pub struct LoggingConfig {
     pub file_logging: Option<PathBuf>,
+    pub opentelemetry_endpoint: Option<Endpoint>,
+}
+impl LoggingConfig {
+    pub fn load(config: &config::Config) -> Result<Self, LoggingConfigError> {
+        let file_logging = None; //TODO load from config
+
+        let opentelemetry_enabled = config.get_bool("monitoring.opentelemetry.enabled")?;
```
[the remark this answers is not in the document]
Comment (Lenita-bu) [reply opendut-util/src/logging.rs:94]: Monitoring was removed to match the name in carl(-development).toml

File: opendut-util/src/logging.rs [hunk 7]
```diff
@@ -46,16 +55,85 @@ pub fn initialize_with_config(config: LoggingConfig) -> Result<(), Error> {
...
+        let opentelemetry_endpoint = if opentelemetry_enabled {
+            let field = String::from("monitoring.opentelemetry.endpoint");
```
[the remark this answers is not in the document]
Comment (Lenita-bu) [reply opendut-util/src/logging.rs:96]: Monitoring was removed to match the name in carl(-development).toml

File: opendut-util/src/logging.rs (hunk 1 above)
Comment (Lenita-bu) [reply opendut-util/src/logging.rs:68]: The parameter for SERVICE_NAME was moved to carl(-development).toml and set to "OTel" there

File: opendut-vpn/opendut-vpn-netbird/src/lib.rs
Comment (Lenita-bu) [reply opendut-vpn/opendut-vpn-netbird/src/lib.rs:1]: tracing macro was added for functions that did not have it.

File: opendut-vpn/opendut-vpn-netbird/src/lib.rs (hunk 2 above)
Comment (Lenita-bu) [reply opendut-vpn/opendut-vpn-netbird/src/lib.rs:72]: The name parameter was removed.

File: opendut-vpn/opendut-vpn-netbird/src/lib.rs (hunk 3 above)
Comment (Lenita-bu) [reply opendut-vpn/opendut-vpn-netbird/src/lib.rs:106]: The name parameter was removed.

File: opendut-carl/carl.toml [hunk 8]
```diff
@@ -33,3 +33,8 @@ auth.type = ""
 auth.secret = ""
 timeout.ms = 10000
 retries = 5
+
+[opentelemetry]
+enabled = false
+endpoint = ""
+service_name = ""
```
Comment (kKdH) [inline opendut-carl/carl.toml:40]: Set service name to `"opendut-carl"`

File: opendut-carl/carl-development.toml [hunk 9]
```diff
@@ -13,3 +13,8 @@ ui.directory = "./opendut-lea/dist/"
 
 [vpn]
 enabled = false
+
+[opentelemetry]
+enabled = true
+endpoint = "http://localhost:4317"
+service_name = "OTel"
```
Comment (kKdH) [inline opendut-carl/carl-development.toml:20]: You can remove the `service_name` config here, because the default set in `carl.toml` is sufficient.

**Review by kKdH — CHANGES_REQUESTED** [review]
my last request for changes :crossed_fingers:

**Review by kKdH — APPROVED** [review]

**Pull request merged**